### PR TITLE
added pgbouncer options

### PR DESCRIPTION
**Pull request by cjestel**

### CODE REVIEW

File: manifests/appuser.pp [hunk 1]
```diff
@@ -20,6 +21,24 @@
       }
     }
 
+    if $pgbouncer_on {
+      unless defined(Pgbouncer::Userlist["cmm_pgsql_module_${username}"]) {
```
Comment (dansajner) [inline manifests/appuser.pp]: We've been using the `ensure_resource` function in lieu of `unless defined` logic in most places. (https://github.com/puppetlabs/puppetlabs-stdlib#ensure_resource)

Comment (dansajner) [inline manifests/appuser.pp]: Don't make that a blocker though if you are trying to move this forward.

Comment (cjestel) [inline manifests/appuser.pp]: It's already scattered throughout this same file other places. I'd rather circle back to address that later if there are no strong objections.

File: manifests/appuser.pp [hunk 2]
```diff
@@ -20,6 +21,24 @@
...
+
+        # create pgbouncer auth_list config
+        pgbouncer::userlist{ "cmm_pgsql_module_${username}":
+          auth_list => [ { user => $username, password => $password }, ],
+        } 
+      }
+
+      unless defined(Pgbouncer::Database["cmm_pgsql_module_${database}_${username}"]) {
```
Comment (everestx) [inline manifests/appuser.pp]: Databases are created in `manifests/appdb.pp`.  Would it be better to keep the call to `pgbouncer::databases` in that manifest?  Unless of course you're going to need to map every user to a database explicitly.

Comment (cjestel) [inline manifests/appuser.pp]: It is indeed the later.  https://github.com/covermymeds/puppet-pgbouncer/blob/master/templates/pgbouncer.ini.databases.part2.erb

Pgbouncer maps based on the database and the user.

File: manifests/appuser.pp [hunk 3]
```diff
@@ -1,5 +1,6 @@
 define cmm_pgsql::appuser (
   $default_handle = false,
+  $pgbouncer_on   = false,
```
Comment (everestx) [inline manifests/appuser.pp]: I would rather use 'enabled', than 'on'.  Also, the flag for `pgbouncer_enabled` should be set in the main class.  Then referenced here as `cmm_pgsql::pgbouncer_enabled`.  Then you don't have to define a parameter in `cmm_pgsql::appuser`.

Comment (cjestel) [inline manifests/appuser.pp]: Do we want it to be a global setting? It's fine if we do, but doing it this way allows us to turn it on only for certain users with hieradata

**Pull request merged**